src/charm.py

```diff
@@ -356,6 +367,21 @@ def _pop_oauth_relation_peer_data(self, relation_id: int) -> Dict:
         key = self._oauth_relation_peer_data_key(relation_id)
         return self._pop_peer_data(key)
 
+    def _get_secrets(self) -> Optional[Dict[str, str]]:
+        try:
+            juju_secret = self.model.get_secret(label=SECRETS_LABEL)
+            return juju_secret.get_content()
+        except SecretNotFoundError:
+            return None
+
+    def _create_secrets(self) -> Optional[Secret]:
+        if not self.unit.is_leader():
+            return None
+
+        secret = {COOKIE_SECRET_KEY: token_hex(16), SYSTEM_SECRET_KEY: token_hex(16)}
+        juju_secret = self.model.app.add_secret(secret, label=SECRETS_LABEL)
```

Comment:
Is it ok to have multiple keys in one juju secret? @gruyaume

Reply:
I think that since they are for the same use, which means that they will most of the time be fetched together and created together, it is fine. 

Now that I think about it though, if rotation is implemented we may want to separate them, so as to allow them to be rotated with different frequency. I think I should split them to different secrets after all.

Reply:
I separated them.